src/main/webapp/js/components/interface/3dCanvas/ThreeDEngine.js

```diff
@@ -137,126 +137,115 @@ define(['jquery'], function () {
             									selected = instancePath;
             									selectedIntersect = intersects[i];
-            									break;
-            								} else if (visible && opacity < 1 && opacity > 0) {
-            									//if only transparent objects intersected select first or the next down if
-            									//one is already selected in order to enable "burrow through" sample.
-            									if (selected == "" && !eval(instancePath + '.selected')) {
-            										selected = instancePath;
-            										selectedIntersect = intersects[i];
-            									} else {
-            										if (eval(instancePath + '.selected') && i != intersects.length - 1) {
-            											selected = "";
-            										}
+            								} else {
+            									if (eval(instancePath + '.selected') && i != intersects.length - 1) {
+            										selected = "";
             									}
             								}
             							}
             						}
             					}
+            				}
 
 
-            					if (selected != "") {
-            						if (that.meshes.hasOwnProperty(selected) || that.splitMeshes.hasOwnProperty(selected)) {
-            							if (!GEPPETTO.isKeyPressed("shift")) {
-            								that.deselectAll();
-            							}
+            				if (selected != "") {
+            					if (that.meshes.hasOwnProperty(selected) || that.splitMeshes.hasOwnProperty(selected)) {
+            						if (!GEPPETTO.isKeyPressed("shift")) {
+            							that.deselectAll();
+            						}
 
-            							var selectedIntersectCoordinates = [selectedIntersect.point.x, selectedIntersect.point.y, selectedIntersect.point.z]
-            							if (geometryIdentifier == undefined) {
-            								geometryIdentifier = "";
-            							}
-            							GEPPETTO.CommandController.execute(selected + '.select(' + false + ', ' + '"' + geometryIdentifier + '", [' + selectedIntersectCoordinates + '])');
+            						var selectedIntersectCoordinates = [selectedIntersect.point.x, selectedIntersect.point.y, selectedIntersect.point.z]
+            						if (geometryIdentifier == undefined) {
+            							geometryIdentifier = "";
             						}
+            						GEPPETTO.CommandController.execute(selected + '.select(' + false + ', ' + '"' + geometryIdentifier + '", [' + selectedIntersectCoordinates + '])');
             					}
-            				} else if (GEPPETTO.isKeyPressed("ctrl")) {
-            					that.deselectAll();
             				}
+            			} else if (GEPPETTO.isKeyPressed("ctrl")) {
+            				that.deselectAll();
             			}
             		}
             	}
             }, false);
 
+
+            this.renderer.domElement.addEventListener('mousemove', function (event) {
+            	//canvas rendered inside whole HTML document
+            	if ( that.renderer.domElement === document ) {
```

Comment:
This if - else I think can be removed and we should be able to take the that.mouse.y and that.mouse.x into the the else as valid in both the scenarios.

e.g.:
that.mouse.y = -((event.clientY -(window.innerHeight -that.renderer.domElement.height))/ that.renderer.domElement.height) * 2 + 1;

if canvas is fullscreen this -(window.innerHeight -that.renderer.domElement.height) should return 0.
Also we use that.renderer.domElement.height to divide the event.clientY, still if canvas is fullscreen this should be equal to window.innerHeight.